templates/reverse_proxy/reverse_proxy.py

```diff
@@ -0,0 +1,59 @@
+from jumpscale import j
+
+from zerorobot.template.base import TemplateBase
+from zerorobot.service_collection import ServiceNotFoundError
+from JumpscaleLib.sal.webgateway.errors import ServiceNotFoundError as WgServiceNotFound
+
+
+WEB_GATEWAY_TEMPLATE_UID = 'github.com/threefoldtech/0-templates/web_gateway/0.0.1'
+
+
+class ReverseProxy(TemplateBase):
+
+    version = '0.0.1'
+    template_name = 'reverse_proxy'
+
+    def __init__(self, name=None, guid=None, data=None):
+        super().__init__(name=name, guid=guid, data=data)
+        self.add_delete_callback(self.uninstall)
+
+    def validate(self):
+        for key in ['webGateway', 'domain', 'servers']:
```

Comment:
An empty servers list is a valid value.
So you can create the reverse_proxy without knowing the backend, then link it to something and this something will configure the backend